feature-wallet-impl/src/main/java/jp/co/soramitsu/feature_wallet_impl/presentation/model/OperationModel.kt

```diff
@@ -0,0 +1,66 @@
+package jp.co.soramitsu.feature_wallet_impl.presentation.model
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import jp.co.soramitsu.feature_wallet_api.R
+import jp.co.soramitsu.feature_wallet_api.domain.model.Operation
+import jp.co.soramitsu.feature_wallet_api.domain.model.Token
+import jp.co.soramitsu.feature_wallet_api.presentation.formatters.formatTokenAmount
+
+class OperationModel(
+    val hash: String,
+    val address: String,
+    val accountName: String?,
+    val type: Operation.Type,
+    val time: Long,
+    val tokenType: Token.Type
+) {
+    val formattedAmount = createFormattedAmount()
+
+    fun getDisplayAddress() = (type as? Operation.Type.Transfer)?.receiver ?: address
+
+    fun getOperationHeader() = format(type.header) ?: accountName ?: getDisplayAddress()
+
+    fun getElementDescription() = format(type.subheader)
+
+    private fun format(extrinsicHeader: String?) = (extrinsicHeader)?.split(regex = "(?<=[a-z])(?=[A-Z])".toRegex())?.joinToString(" ")?.capitalize()
+
+    fun getOperationIcon(): Int? = when (type) {
+        is Operation.Type.Reward -> R.drawable.ic_staking
+        else -> null
+    }
+
+    fun getIsIncome() = when (type) {
+        is Operation.Type.Extrinsic -> false
+        is Operation.Type.Reward -> type.isReward
+        is Operation.Type.Transfer -> address == type.receiver
+    }
+
+    val amountColorRes = when {
+        type.status == Operation.Status.FAILED -> R.color.gray2
+        getIsIncome() -> R.color.green
+        else -> R.color.white
+    }
+
+    val statusAppearance = when (type.status) {
+        Operation.Status.COMPLETED -> StatusAppearance.COMPLETED
+        Operation.Status.FAILED -> StatusAppearance.FAILED
+        Operation.Status.PENDING -> StatusAppearance.PENDING
+    }
```

Comment:
I still suggest putting this logic inside the mapOperationToOperationModel
The reason is that execution of a mapper is done by viewmodel on background thread, whereas fields are getting accessed by view on the main thread
Also, having ui model only contains fields that will actually be used by view provides more structure (compare with  having several fields, that are only used for calculating other fields)